src/piqtree2/iqtree/_tree.py

```diff
@@ -9,19 +9,77 @@
 
 from piqtree2.exceptions import ParseIqTreeError
 from piqtree2.iqtree._decorator import iqtree_func
-from piqtree2.model import Model
+from piqtree2.model import DnaModel, Model
 
 iq_build_tree = iqtree_func(iq_build_tree, hide_files=True)
 iq_fit_tree = iqtree_func(iq_fit_tree, hide_files=True)
 
+# the order defined in IQ-TREE
+RATE_PARS = "A/C", "A/G", "A/T", "C/G", "C/T", "G/T"
+MOTIF_PARS = "A", "C", "G", "T"
+
 
 def _rename_iq_tree(tree: cogent3.PhyloNode, names: Sequence[str]) -> None:
     for tip in tree.tips():
         tip.name = names[int(tip.name)]
 
 
-def _process_tree_yaml(tree_yaml: dict, names: Sequence[str]) -> cogent3.PhyloNode:
+def _insert_edge_pars(tree: cogent3.PhyloNode, **kwargs: dict) -> None:
+    # inserts the edge parameters into each edge to match the structure of
+    # cogent3.PhyloNode
+    for node in tree.get_edge_vector():
+        # skip the rate parameters when node is the root
+        if node.is_root():
+            kwargs = {k: v for k, v in kwargs.items() if k == "mprobs"}
+            del node.params["edge_pars"]
+        node.params.update(kwargs)
+
+
+def _format_edge_pars(tree: cogent3.PhyloNode, model: Model) -> None:
+    # converts the strings of rate and motif parameters into a dictionary format
+    rate_pars = dict(
+        zip(RATE_PARS, map(float, tree.params["edge_pars"]["rates"].split(", "))),
+    )
+    motif_pars = {
```

Comment:
I think all type casting to float / int should be done within `_process_tree_yaml()`.